tools/travis_watchdog.sh

```diff
@@ -177,6 +177,14 @@ print_stacktraces () {
 	done
 }
 
+collect_coredumps() {
+	echo "Searching for .dump, .dumpstream and related files in $($HERE/../)"
+	for file in `find . -type f -regextype posix-extended -iregex '.*\.dump|.*\.dumpstream|.*hs.*\.log|.*/core(.[0-9]+)?$'`; do
```

Comment:
a simpler solution may be to setup the JVM to directly write the dumps into `ARTIFACTS_DIR`.
https://stackoverflow.com/questions/1880166/is-it-possible-to-specify-where-jvms-crash-dumps-go

Reply:
With this implementation, we can use this function also for collecting any JVM debugging files from the end to end tests.
Yes, we could also inject JVM properties through environment variables into most of the end to end tests. But who guarantees that Kafka, ElasticSearch, Zookeeper all properly respect the environment variables.

With this approach, I'm following a blackbox approach where I don't trust the system I'm testing.

Also, I don't know if I would spend my time well for the project if I would refactor something that works. Coredumps (luckily) happen quite rarely. I just wanted to make life for others easier when they have to debug a crashing JVM on a CI system.

Reply:
> But who guarantees that Kafka, ElasticSearch, Zookeeper all properly respect the environment variables.

There's no guarantee that these are run under FLINK_ROOT though; the java e2e tests use a junit temporary folder to store and run these services, as well as the flink distribution.
At least you'd also have to search the entire tmp directory of the system, or modify `java.io.tmpdir` to point somewhere under FLINK_ROOT.

Have you tried the `JAVA_TOOL_OPTIONS` environment variable that the JVM itself evaluates? https://docs.oracle.com/javase/8/docs/technotes/guides/troubleshoot/envvars002.html

> I don't know if I would spend my time well for the project if I would refactor something that works.

This can be read in rather frightening ways: "Lol let's use the first version that works _now_."